Neos.Workspace.Ui/Classes/Controller/WorkspaceController.php

```diff
@@ -1013,7 +1027,7 @@ protected function prepareBaseWorkspaceOptions(
         ContentRepository $contentRepository,
         WorkspaceName $excludedWorkspace = null,
     ): array {
-        $user = $this->userService->getCurrentUser();
+        $authenticatedAccount = $this->securityContext->getAccount();
```

Comment:
That's right. The possibility to attach _multiple_ accounts to a single Neos user was added sometimes in the past. But I doubt that it is used _anywhere_ in practice because it has some fundamental logic flaws.
For example: `UserService::getAllRoles()` returns all roles of all attached accounts. But there is always only one authenticated account at a time, so that combination of roles does not really help..

Reply:
This is just half the problem, I think the more possible scenario is that you have frontend users and logged in with one of those AND your backend account. I this case you want to ensure in places like this here that you get the right account. Yes I know people will _probably_ not have their frontend users in the request patterns for backend, but I have seen it and I think it would get hard to debug and understand if it goes wrong even if that is a rare case.
tl;dr; I really think we should keep the existing logic.

Reply:
What would the existing logic be in this case?

Reply:
Iterated authenticated accounts and find the first that has a neos user attached

Reply:
> Iterated authenticated accounts and find the first that has a neos user attached

I see. I was afraid that you'd suggest to change the whole API.
Unfortunately the security context does not allow to iterate authenticated accounts, but what about 8b8eea7700669e1edaece155d6ebbfc60a39022a ?

Reply:
Uiuiuiuuiu what have we gotten ourselves into 😭 😅 ... maybe that direction?

But you forgot to change it at all places: https://github.com/neos/neos-development-collection/blob/41e6b57ba8917ef394e97e4ab511db37453032d6/Neos.Neos/Classes/Security/ContentRepositoryAuthProvider/ContentRepositoryAuthProvider.php#L202

Reply:
Thanks basti for simplifying and fixing this via https://github.com/neos/neos-development-collection/pull/5298/commits/466c89e583dbbc9bfdf356e256f7f5cea69ec28f ❤️